src/libMessage/Messenger.cpp

```diff
@@ -111,6 +115,227 @@ void NumberToArray(const T& number, vector<unsigned char>& dst,
+  account.SetBalance(tmpNumber);
+
+  ProtobufByteArrayToNumber<uint256_t, UINT256_SIZE>(protoAccount.nonce(),
+                                                     tmpNumber);
+  account.SetNonce(tmpNumber);
+
+  dev::h256 tmpStorageRoot;
+  copy(protoAccount.storageroot().begin(),
+       protoAccount.storageroot().begin() +
+           min((unsigned int)protoAccount.storageroot().size(),
+               (unsigned int)tmpStorageRoot.size),
+       tmpStorageRoot.asArray().begin());
+
+  if (protoAccount.code().size() > 0) {
+    vector<unsigned char> tmpVec;
+    tmpVec.resize(protoAccount.code().size());
+    copy(protoAccount.code().begin(), protoAccount.code().end(),
+         tmpVec.begin());
+    account.SetCode(tmpVec);
+
+    dev::h256 tmpHash;
+    copy(protoAccount.codehash().begin(),
+         protoAccount.codehash().begin() +
+             min((unsigned int)protoAccount.codehash().size(),
+                 (unsigned int)tmpHash.size),
+         tmpHash.asArray().begin());
+
+    if (account.GetCodeHash() != tmpHash) {
+      LOG_GENERAL(WARNING,
+                  "Code hash mismatch. Expected: "
+                      << DataConversion::charArrToHexStr(
+                             account.GetCodeHash().asArray())
+                      << " Actual: "
+                      << DataConversion::charArrToHexStr(tmpHash.asArray()));
+      return false;
+    }
+
+    account.SetCreateBlockNum(protoAccount.createblocknum());
+
+    if (protoAccount.initdata().size() > 0) {
+      tmpVec.resize(protoAccount.initdata().size());
+      copy(protoAccount.initdata().begin(), protoAccount.initdata().end(),
+           tmpVec.begin());
+      account.InitContract(tmpVec);
+    }
+
+    for (const auto& entry : protoAccount.storage()) {
+      copy(entry.keyhash().begin(),
+           entry.keyhash().begin() + min((unsigned int)entry.keyhash().size(),
+                                         (unsigned int)tmpHash.size),
+           tmpHash.asArray().begin());
+      account.SetStorage(tmpHash, entry.data());
+    }
+
+    if (account.GetStorageRoot() != tmpStorageRoot) {
+      LOG_GENERAL(WARNING, "Storage root mismatch. Expected: "
+                               << DataConversion::charArrToHexStr(
+                                      account.GetStorageRoot().asArray())
+                               << " Actual: "
+                               << DataConversion::charArrToHexStr(
+                                      tmpStorageRoot.asArray()));
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void AccountDeltaToProtobuf(const Account* oldAccount,
+                            const Account& newAccount,
+                            ProtoAccount& protoAccount) {
+  Account acc(0, 0);
+
+  bool fullCopy = false;
+
+  if (oldAccount == nullptr) {
+    oldAccount = &acc;
+    fullCopy = true;
+  }
+
+  int256_t balanceDelta =
+      int256_t(newAccount.GetBalance()) - int256_t(oldAccount->GetBalance());
```

Comment:
Can we use safemath here?

Reply:
On second thought, will not change this for now since SafeMath doesn't support signed type. Issue created: https://github.com/Zilliqa/Zilliqa/issues/792